### DISCUSSION TITLE
TimeStepper not updating dt after --recover

### DISCUSSION THREAD

**Discussion by KhaledNabilSharafeldin** [idaholab/moose#21131] [Q&A General] [answered]
I am currently running a long run with checkpoints enabled. When the run is stopped, and then recovered using the flag `--recover` in the command line it starts back up again at the last checkpoint and starts solving seemingly fine. 
However, the problem is that the TimeStepper that updates dt based on postprocessor information does not update the dt anymore, and is constant at the initial guess of dt.

is there a specific reason why this is the case? and how can I enable it?

Thanks in advance

**Comment by GiudGiud** [comment, marked as answer]
Hello

I just checked on the test we have for it and it works.
The time stepper should not have changed, it should keep using the postprocessor value. 

What does your executioner block look like ?

Guillaume

**Reply by KhaledNabilSharafeldin** [reply]
I have multiple subapps

this is the main
```
[Executioner]
  type = Transient
  solve_type = 'PJFNK'    # PJFNK JFNK NEWTON FD LINEAR
  petsc_options_iname = '-pc_type -pc_hypre_type -ksp_gmres_restart -pc_hypre_boomeramg_strong_threshold'
  petsc_options_value = 'hypre    boomeramg      31                 0.7'
  nl_rel_tol = 1e-10  # first iteration tolerance/current
  nl_abs_tol = 1e-5 # current iteration tolerance
  l_max_its = 1000
  num_steps = 100000
  start_time = 5000
  # dtmin = .1
  [./TimeStepper]
    type = TransientTransport
    postprocessor = rss_max
    cfl = 0.45
    dt = 1e-6
    dt_max = 50
    n = 20
    n_slip = 12
  [../]  
[]
```

1st subapp

```
[Executioner]
  type = Transient
  solve_type = 'LINEAR'                  # PJFNK JFNK NEWTON FD LINEAR
  l_tol = 1.0e-10                         # Linear Tolerance         : 1.0e-5
  l_abs_tol = 1.0e-8
  l_max_its = 10000                       # Max Linear Iterations    : 10000
  petsc_options_iname = '-pc_type -pc_hypre_type -ksp_gmres_restart -pc_hypre_boomeramg_strong_threshold'
  petsc_options_value = 'hypre    boomeramg      31                 0.7'
  [./TimeStepper]
    type = TransientTransport
    postprocessor = rss_max
    cfl = 0.45
    dt = 1e-6
    dt_max = 50
    n_slip = 12
  [../] 
  num_steps = 1000

  automatic_scaling = true
  compute_scaling_once = true 
[]
```

2nd subapp
```
[Executioner]
  type = Transient
  solve_type = 'LINEAR'                  # PJFNK JFNK NEWTON FD LINEAR
  l_tol = 1.0e-11                         # Linear Tolerance         : 1.0e-5
  l_max_its = 10000                       # Max Linear Iterations    : 10000
  petsc_options_iname = '-pc_type   -pc_hypre_type  -pc_hypre_boomeramg_strong_threshold'
  petsc_options_value = 'hypre      boomeramg       0.7'

  [./TimeIntegrator]
    # type = ImplicitEuler
    # type = BDF2
    #type = CrankNicolson
    # type = ImplicitMidpoint
    # type = LStableDirk2
    # type = LStableDirk3
    # type = LStableDirk4
    # type = AStableDirk4
    #
    # Explicit methods
    type = ActuallyExplicitEuler
    # type = ExplicitEuler
    #type = ExplicitMidpoint
    # type = Heun
    # type = Ralston
  [../]
  [./TimeStepper]
    type = TransientTransport
    postprocessor = rss_maxs
    cfl = 0.45
    dt = 1e-6
    dt_max = 50
    n_slip = 12
  [../] 
  # scheme = implicit-euler
[]
```

pardon the long reply.

The time stepping seems to be working fine when it is launched properly, and stops functioning when I recover it. by stops working I mean it uses the initial dt=1e-6 all the time. 
This timestepper is custom, and I have it print out the calculated dt in the terminal, so it does calculate a new dt, but is not used.

let me know if I need to provide more details.

**Reply by GiudGiud** [reply]
So when you print the time step, do all 3 timesteppers return something bigger than 1e-6 ?

The main app is usually the one that imposes the shortest timestep, though the subapps may be subcycling.

I think at this point I would want to see the log of the execution.

**Reply by GiudGiud** [reply]
@KhaledNabilSharafeldin any update on this?
